File: pkg/dev_compiler/test/codegen/expect/sunflower/painter.js
dart_library.library('sunflower/painter', null, /* Imports */[
  'dart/_runtime',
  'dart/math',
  'sunflower/dom',
  'dart/core',
  'sunflower/circle'
], /* Lazy imports */[
], function(exports, dart, math, dom, core, circle) {
  'use strict';
  let dartx = dart.dartx;
  const ORANGE = "orange";
  const RED = "red";
  const BLUE = "blue";
  const TAU = dart.notNull(math.PI) * 2;
  function querySelector(selector) {
    return dom.document.querySelector(selector);
  }
  dart.fn(querySelector, dom.Element, [core.String]);
  dart.defineLazyProperties(exports, {
    get canvas() {
      return dart.as(querySelector("#canvas"), dom.CanvasElement);
    }
  });
  dart.defineLazyProperties(exports, {
    get context() {
      return dart.as(exports.canvas.getContext('2d'), dom.CanvasRenderingContext2D);
    }
  });
  class CirclePainter extends core.Object {
    CirclePainter() {
      this.color = ORANGE;
    }
    draw(context) {
      context.beginPath();
      context.lineWidth = 2;
      context.fillStyle = this.color;
      context.strokeStyle = this.color;
      context.arc(this.x, this.y, this.radius, 0, TAU, false);
      context.fill();
      context.closePath();
      context.stroke();
    }
  }
  CirclePainter[dart.implements] = () => [circle.Circle];
  dart.setSignature(CirclePainter, {
    methods: () => ({draw: [dart.void, [dom.CanvasRenderingContext2D]]})
  });
  // Exports:
  exports.ORANGE = ORANGE;
  exports.RED = RED;
  exports.BLUE = BLUE;
  exports.TAU = TAU;
  exports.querySelector = querySelector;
  exports.CirclePainter = CirclePainter;
});

File: pkg/dev_compiler/test/codegen/sunflower/painter.dart
// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

library painter;

import 'dart:html';
import 'dart:math';

import 'circle.dart';

const ORANGE = "orange";
const RED = "red";
const BLUE = "blue";
const TAU = PI * 2;

Element querySelector(String selector) => document.querySelector(selector);

final canvas = querySelector("#canvas") as CanvasElement;
final context = canvas.getContext('2d') as CanvasRenderingContext2D;

abstract class CirclePainter implements Circle {
  // This demonstrates a field in a mixin.
  String color = ORANGE;

  /// Draw a small circle representing a seed centered at (x,y).
  void draw(CanvasRenderingContext2D context) {
    context
      ..beginPath()
      ..lineWidth = 2
      ..fillStyle = color
      ..strokeStyle = color
      ..arc(x, y, radius, 0, TAU, false)
      ..fill()
      ..closePath()
      ..stroke();
  }
}
